### ISSUE TITLE
Repeater field returning false instead of [] when empty

### ISSUE THREAD

**Issue by henk23** [AdvancedCustomFields/acf#701]
This is an excerpt of one of our acf-json files:

    {
      "key": "field_1111,
      "label": "Filters",
      "name": "filters",
      "type": "repeater",
      "sub_fields": [
        {
          "key": "field_2222",
          "label": "Key",
          "name": "key",
          "type": "select",
          "choices": {
            "foo": "Foo",
            "bar": "Bar"
          },
          "default_value": [], // <-- this changed to false after ACF Pro update from 5.8.7 to 5.12.3
          "allow_null": 0,
          "multiple": 0,
          "return_format": "value",
        }
      ]
    }

And sure, this makes sense. The sub_field "key" is a single select so it should return false when nothing is selected. In case of multiple select, an empty array makes more sense.

But: In case the parent repeater field "filters" is empty (= no items inside the repeater) ACF Pro will return `false` for this field. Expectation should still be an empty array for an empty repeater. Right?

**Comment by CreativeDive** [comment]
@henk23 I've noticed this in the past too. I always solved it in this case with:

`$field_value = is_array( $field ) ? $field : array(); `

**Comment by montchr** [comment]
Yeah `false` is a pretty terrible default… even `null` would be better. Nullable types are a thing, but there's no such thing as falsifiable types.

**Comment by raray** [comment]
Would love to see a response to this. It's a pretty major miss.

**Comment by salmanamemon** [comment]
Yes i am exposing it with restapi however the same issue i am facing because when doing headless you have to define in typescript if it is string, boolean etc and for gallery false is breaking it.
